clean-abap/CleanABAP_es.md

```diff
@@ -0,0 +1,5123 @@
+
+```ABAP
+ASSIGN generic->* TO FIELD-SYMBOL(<generic>).
+ASSIGN COMPONENT name OF STRUCTURE structure TO FIELD-SYMBOL(<component>).
+ASSIGN (class_name)=>(static_member) TO FIELD-SYMBOL(<member>).
+```
+
+Las revisiones de código demuestran que la gente tiende a escoger entre las dos arbitrariamente,
+"solo porque sí", "porque siempre hacemos LOOPs de esa manera", o "sin una razón en particular".
+Las decisiones arbitrarias hacen que el lector pierda el tiempo en la pregunta sin sentido de por qué
+se usó uno en lugar del otro, por lo tanto se debería reemplazar con decisiones precisas y bien fundamentadas.
+Nuestra recomendación está basada en el siguiente razonamiento:
+
+- Los `field-symbols` pueden hacer cosas que las referencias no pueden, como acceder dinámicamente a los componentes de una estructura.
+De la misma manera, las referencias pueden hacer cosas que los `field-symbols` no, como construir dinámicamente una estructura de datos con tipos. En resumen, elegir una sola para todas las situaciones no es posible.
+
+- En ABAP orientado a objetos, las referencias se usan por todas partes y no se pueden evitar, ya que cualquier
+objeto es una `REF TO <class-name>`. 
+En contraste, los `field-symbols` son solamente requeridos estrictamente en situaciones especiales al usar
+programación dinámica. Las referencias por lo tanto tienen una preferencia natural en cualquier programa
+orientado a objetos.
+
+- Los `field-symbols` son más cortos que las referencias, pero el ahorro resultante en memoria es tan
+pequeño que puede ser ignorado sin problema.
+De la misma manera, la velocidad no es un problema. Como consecuencia, no hay razón relacionada con el rendimiento
+para preferir uno sobre el otro.
+
+> Lee más en el artículo
+> [_Accessing Data Objects Dynamically_ in the ABAP Programming Guidelines](https://help.sap.com/doc/abapdocu_751_index_htm/7.51/en-US/index.htm?file=abendyn_access_data_obj_guidl.htm).
+
+## Tablas
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Esta sección](#tablas)
+
+### Usa el tipo correcto de tabla
+
+> [Clean ABAP](#clean-abap) > [Contenido](#contenido) > [Tablas](#tablas) > [Esta sección](#usa-el-tipo-correcto-de-tabla)
+
+- Típicamente se usan tablas del tipo `HASHED` para **tablas grandes** que son 
+**llenadas en un solo paso**, **nunca se modifican** y son **leídas seguido por su llave**.
+Su overhead en memoria y procesamiento hace a las tablas hash únicamente valiosas
+cuando se tienen cantidades grandes de datos y muchos accesos de lectura.
+Cada cambio al contenido d ela tabla requiere costosos recálculos del hash,
+así que no uses este tipo para tablas que son modificadas muy seguido.
+
+- Típicamente se usan tablas del tipo `SORTED` para **tablas grandes**
+que necesitan estar **ordenadas en todo momento**, que son **llenadas poco a poco** o que 
+**necesitan ser modificadas** y **leídas por una o más llaves parciales o completas** o procesadas
```

Comment:
```suggestion
**necesitan ser modificadas** y **leídas por una o más claves parciales o completas** o procesadas
```

Reply:
In Latin America _llave_ is always used to refer to DB keys, but I did a quick Google search for both _"llave primaria"_ and _"clave primaria"_ and it seems the latter is way more used. I'll keep the suggestion.